ql/indexes/equityindex.hpp

```diff
@@ -0,0 +1,119 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Marcin Rybacki
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file equityindex.hpp
+    \brief base class for equity indexes
+*/
+
+#ifndef quantlib_equityindex_hpp
+#define quantlib_equityindex_hpp
+
+#include <ql/index.hpp>
+#include <ql/time/calendar.hpp>
+#include <ql/currency.hpp>
+#include <ql/termstructures/yieldtermstructure.hpp>
+
+namespace QuantLib {
+
+    //! Base class for equity indexes
+    /*! The equity index object allows to retrieve past fixings,
+        as well as project future fixings using either both
+        the risk free interest rate term structure and the dividend
+        term structure, or just the interest rate term structure
+        in which case one can provide a term structure of equity
+        forwards implied from, e.g. option prices.
+        
+        In case of the first method, the forward is calculated as:
+        \f[
+        I(t, T) = I(t, t) \frac{P_{D}(t, T)}{P_{R}(t, T)},
+        \f]
+        where \f$ I(t, t) \f$ is today's value of the index,
+        \f$ P_{D}(t, T) \f$ is a discount factor of the dividend
+        curve at future time \f$ T \f$, and \f$ P_{R}(t, T) \f$ is
+        a discount factor of the risk free curve at future time
+        \f$ T \f$.
+
+        In case of the latter method, the forward is calculated as:
+        \f[
+        I(t, T) = I(t, t) \frac{1}{P_{F}(t, T)},
+        \f]
+        where \f$ P_{F}(t, T) \f$ is a discount factor of the equity
+        forward term structure.
+
+        To forecast future fixings, the index requires today's fixing.
+    */
+    class EquityIndex : public Index, public Observer {
+      public:
+        EquityIndex(std::string name,
+                    Currency currency,
```

Comment:
I'm not sure about having the currency here.  True, we added it to some other indexes, but it's pretty much unused.  It might just be extra baggage.  What do you think?

Reply:
Right, currency is not really used. I will remove it.